src/traits.rs

```diff
@@ -0,0 +1,26 @@
+pub trait Device {
+    fn mapping_def(&self) -> std::ops::Range<usize>;
+
+    fn map(&mut self, memory: &mut [u8]);
```

Comment:
I feel like we should try to pass a slice to a relevant memory and not an entire array + address range

Reply:
I the code I'm passing a slice. Not the entire array of memory